tikv-client/src/main/java/com/pingcap/tikv/operation/iterator/ScanIterator.java

```diff
@@ -110,21 +108,21 @@ public boolean hasNext() {
     if (isCacheDrained()) {
       return null;
     }
-    if (index < currentCache.size()) {
-      return currentCache.get(index++);
-    }
-    return null;
+    return currentCache.get(index++);
   }
 
   @Override
   public Kvrpcpb.KvPair next() {
+    if (currentCache == null) {
+      throw new TiClientInternalException("cache is null, read data failed");
```

Comment:
It seems not necessary. We already checked `currentCache`'s value at `isCacheDrained`